starter-code/src/App.js

```diff
@@ -1,18 +1,74 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+// import logo from "./logo.svg";
+import "./App.css";
+import contacts from "./contacts.json";
+import Contacts from "./Contacts";
 
 class App extends Component {
+  state = {
+    contactsLists: contacts.slice(0, 5)
+  };
+
+  addRandomContact = () => {
+    // here we find a random contact
+    // and we add it to the state for contactsList
+
+    const random = contacts[Math.floor(Math.random() * contacts.length)];
+
+    console.log(random);
+
+    this.setState({
+      contactsLists: [random, ...this.state.contactsLists]
```

Comment:
Keep in mind that you never check if the contact is duplicated meaning you may get an error when you set your Contact Component key={id} since this id may be similar for some of the contacts